src/utilities/geometry/Polygon3d.cpp

```diff
@@ -0,0 +1,236 @@
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include "Polygon3d.hpp"
+#include "Vector3d.hpp"
+#include "Geometry.hpp"
+#include "Intersection.hpp"
+#include <utilities/geometry/Transformation.hpp>
+
+namespace openstudio {
+
+Polygon3d::Polygon3d() {}
+
+Polygon3d::Polygon3d(const Point3dVector& outerPath) {
+  for (const auto& p : outerPath) {
+    m_outerPath.emplace_back(p);
+  }
+}
+
+Polygon3d::Polygon3d(const Point3dVector& outerPath, const Point3dVectorVector& innerPaths) {
+  for (const auto& p : outerPath) {
+    m_outerPath.emplace_back(p);
+  }
+
+  for (const auto& innerPath : innerPaths) {
+    addHole(innerPath);
+  }
+}
+
+/// Adds a point to the polygon perimeter
+void Polygon3d::addPoint(const Point3d& point) {
+  m_outerPath.emplace_back(point);
+}
+
+/// Sets the perimeter of the polygon
+void Polygon3d::setOuterPath(const Point3dVector& outerPath) {
+  m_outerPath = outerPath;
+}
+
+Point3dVector Polygon3d::getOuterPath() const {
+  return m_outerPath;
+}
+
+Point3dVectorVector Polygon3d::getInnerPaths() const {
+  return m_innerPaths;
+}
+
+void Polygon3d::addHole(const Point3dVector& hole) {
+  m_innerPaths.emplace_back(hole);
+}
+
+Vector3d Polygon3d::newallVector() {
+  if (OptionalVector3d v = openstudio::getNewallVector(m_outerPath)) {
+    return v.get();
+  }
+
+  return Vector3d();
+}
+
+Vector3d Polygon3d::outwardNormal() {
+  return openstudio::getOutwardNormal(m_outerPath).get();
+}
+
+double Polygon3d::grossArea() {
+
+  if (boost::optional<double> area = openstudio::getArea(m_outerPath)) {
+    return area.get();
+  }
+
+  return 0;
+}
+
+double Polygon3d::netArea() {
+
+  double netArea = grossArea();
+  for (const auto& hole : m_innerPaths) {
+    if (boost::optional<double> area = openstudio::getArea(hole)) {
+      netArea -= area.get();
+    }
+  }
+
+  return netArea;
+}
+
+double Polygon3d::perimeter() {
+
+  double perimeter = 0;
+  for (size_t i = 0; i < m_outerPath.size(); i++) {
+    Point3d p1 = m_outerPath[i];
+    Point3d p2 = m_outerPath[(i + 1) % m_outerPath.size()];
+    perimeter += openstudio::getDistance(p1, p2);
+  }
+
+  return perimeter;
+}
+
+bool Polygon3d::isClockwise() {
+  if (OptionalVector3d normal = getOutwardNormal(m_outerPath)) {
```

Comment:
This doesn't really make sense, it's only clockwise if viewed from above or something, do you need this method? I think it might be backwards too.  At least as I think about it a clockwise polygon has an outward normal that goes down.

Reply:
You are right it is reversed (well with RHS coordinates it is). On reflection I think it would be best to remove this method